xbmc/video/VideoInfoScanner.cpp

```diff
@@ -128,36 +128,47 @@ namespace KODI::VIDEO
       // result in unexpected behaviour.
       m_bCanInterrupt = false;
 
-      bool bCancelled = false;
-      while (!bCancelled && !m_pathsToScan.empty())
+      if (m_handle)
+        m_handle->SetTitle(g_localizeStrings.Get(20475));
+
+      // List all media files of all 'm_pathsToScan'
+      //   INFO: m_pathsToScan may contain path with different library content, so we store their properties into a dedicated struct.
+      std::vector<SourcePathContent> _pathsContent;
+
+      for (const auto& rootPathToScan : m_pathsToScan)
       {
-        /*
-         * A copy of the directory path is used because the path supplied is
-         * immediately removed from the m_pathsToScan set in DoScan(). If the
-         * reference points to the entry in the set a null reference error
-         * occurs.
-         */
-        std::string directory = *m_pathsToScan.begin();
-        if (m_bStop)
+        // Load root path properties from scraper settings:
+        bool foundDirectly = false;
+        SScanSettings rootPathScraperScanSettings;
+        const ScraperPtr info = m_database.GetScraperForPath(
+            rootPathToScan, rootPathScraperScanSettings, foundDirectly);
+        const CONTENT_TYPE contentType = info ? info->Content() : CONTENT_NONE;
+
+        // List and store medias:
+        if (IsPathAllowedForScan(rootPathToScan, rootPathScraperScanSettings, contentType))
         {
-          bCancelled = true;
+          SourcePathContent pathContent = {rootPathToScan, rootPathScraperScanSettings,
+                                           contentType};
+          pathContent.items.SetPath(
+              rootPathToScan); // We use the 'path' of CFileItemList to store its root-path-to-scan
+
+          LoadRemoteItems(rootPathToScan, pathContent.items, rootPathScraperScanSettings,
+                          contentType);
+
+          _pathsContent.push_back(pathContent);
         }
-        else if (!CDirectory::Exists(directory))
+
+        // Support scan cancellation
+        if (m_bStop)
         {
-          /*
-           * Note that this will skip clean (if m_bClean is enabled) if the directory really
-           * doesn't exist rather than a NAS being switched off.  A manual clean from settings
-           * will still pick up and remove it though.
-           */
-          CLog::Log(LOGWARNING, "{} directory '{}' does not exist - skipping scan{}.", __FUNCTION__,
-                    CURL::GetRedacted(directory), m_bClean ? " and clean" : "");
-          m_pathsToScan.erase(m_pathsToScan.begin());
+          break;
         }
-        else if (!DoScan(directory))
-          bCancelled = true;
       }
 
-      if (!bCancelled)
+      m_pathsToScan.clear();
```

Comment:
Looks like that would interfere with the "cleanonupdate" advanced setting that sets m_bClean. I'm not sure of the details but from a high level, after a scan, the clean likely removes from the library the movies/tv shows that were not found in the sources because they were deleted on disk (whatever was left in m_pathsToScan after a scan, with current algorithm)